### ISSUE TITLE
Custom events by ember-gestures are working in testing

### ISSUE THREAD

**Issue by RVenkatesh** [emberjs/ember-test-helpers#402]
I am writing a ember component [ember-slider](https://github.com/RVenkatesh/Ember-Slider). I wrote some basic test cases for the component. But when I tried to write a test case triggering 'tap' event using ember-test-helpers, the event was not recognised by "ember-gestures" add-on which I am using  capture user actions like tap and pan.

Following is the code I am using to test the tap functionality. Please help me figure out what is going wrong. I am not sure if I have provided enough detail here. Please let me know in case if more required.

```
test('tap to change value', async function(assert) {
     assert.expect(2);
     await render(hbs`{{ember-slider initialValue=10}}`);
     assert.equal(parseInt(this.$('.slider-value')[0].textContent), 10, 'Value set to 10');
    
     // The following jQuery event callback is called after tap('.ember-slider'); gets executed. But the tap event inside my component is not.
     this.$('.ember-slider').on('click', function(e) {
       console.log("Entered ", e);
     });
     
     await tap('.ember-slider');
     assert.equal(parseInt(this.$('.slider-value')[0].textContent), 0, 'Value set back to 0 by tapping');
});

**Comment by rwjblue** [comment]
Unfortunately, it doesn't look like ember-gestures has meaningful tests around this behavior, and I'm no expert on how that library works.

Can you provide a minimal test scenario that somewhat correctly demonstrates the failure in your app?  If you do it then folks can use it to deep dive into the issue you are facing, but without a repro it is pretty unlikely that we will be able to help you...

I'm thinking something like:

```
ember new tap-event-failure-testing
cd tap-event-failure-testing
ember install ember-gestures
ember generate component my-tap-aware-component
```

Then modify both the component and the test to emulate the real world things in your app...

**Comment by rwjblue** [comment]
OK, @rondale-sc and I spent some time poking at this this morning. Somewhat random brain dump follows....

* We created a repo to show the failure (mostly following from my last comment) [here](https://github.com/rwjblue/__tap-event-testing).
* Hammer.js uses fallback logic to determine which events to hook into and use for firing the `tap` event. That fallback logic is [here](https://github.com/hammerjs/hammer.js/blob/a28c530297f6e66785476013a6c9162305a2ce42/src/inputjs/create-input-instance.js#L15-L31).
* The tests in that repo pass on Safari (does not support `PointerEvent`) but not on Chrome (supports `PointerEvent`).
* The internal `fireEvent` helper function is not aware of `PointerEvent`s, and for `touchstart` / `touchend` events we fire the generic `Event`.

Action items:

- [ ] Add support for [touch events](https://developer.mozilla.org/en-US/docs/Web/API/Touch_events) to the fire event helper
  - Use `PointerEvent` when `window.PointerEvent` is present and fallback otherwise
- [ ] Add Hammer.js as a dev dependency to this repo (either via ember-gestures, or just as an `app.import`) along with a smoke test like the one written in the demo repo above to confirm tap continues to work with Hammer.js.

**Comment by RVenkatesh** [comment]
Hey! Thank you very much for the quick reply. I built a [Ember twiddle](https://ember-twiddle.com/9a62ca3e567e413f1d8143ab4b792480?openFiles=tests.integration.components.my-component-test.js%2Ctemplates.components.my-component.hbs). Below given are the details about this twiddle.

`my-component ` inserts text on click and tap events.
`my-component-test` contains the test for both click and tap.

Note that the test for click passes and tap does not. Also while manually doing the same, both click and tap work.

**Second thoughts**
While creating this twiddle, I realised this should be a bug related to ember-gestures than ember-test-helpers. If you also think the same, let me know. I will report this issue there.

**Comment by RVenkatesh** [comment]
You are right. The tests pass in Safari. Only in Chrome they don't. Thank you once again.

**Comment by eriktrom** [comment]
i'll check this out in more detail but try ember-gestures 1.0.0 in the meantime, if that works, then 
this change in 1.1.0 is why - https://github.com/html-next/ember-gestures/pull/114/files (probably lines 156-164 in split diff mode, right side)

**Comment by RVenkatesh** [comment]
@eriktrom I checked it with 1.0.0 version. The issue is still there. May be it is related to what @rwjblue mentioned.
